### ISSUE TITLE
Fase 1 (FD001) - Pipeline completo con MLflow tracking para FD001

### ISSUE THREAD

**Issue by JairRamosG** [JairRamosG/RUL-CMAPSS#9]
## Goal

Integrar todos los módulos desacoplados del pipeline (`data`, `features`, `models`, `evaluation`) en un script ejecutable (`scripts/train_eval.py`) que orqueste el flujo experimental completo bajo un esquema estricto de validación cruzada agrupada (`GroupKFold` con k=10), perfilando recursos computacionales (tiempo de entrenamiento, latencia de inferencia y consumo de memoria RAM pico) y registrando automáticamente parámetros, métricas y el análisis de inferencia estadística multimodelo (Friedman + Nemenyi / RM-ANOVA) como artefactos en **MLflow**.

### 1. Secuencia Canónica del Pipeline (Blindaje Anti-Leakage)
El flujo de datos garantiza que no exista fuga de información entre particiones:
```
1. Carga de datos crudos (train_FD001.txt, test_FD001.txt, RUL_FD001.txt)
2. Filtrado de 7 sensores constantes identificados en el EDA
3. Etiquetado Piecewise Linear RUL con saturación en RUL_max = 125 ciclos
4. Feature Engineering temporal (Rolling stats y tendencias/deltas por motor)
5. Partición GroupKFold (k=10) sobre unit_number de train (10 motores por fold exactos)
   └── Por cada Fold:
       ├── Ajustar MinMaxScaler exclusivamente sobre X_train_fold
       ├── Transformar X_train_fold y X_val_fold
       ├── Estructurar secuencias temporales en ventanas 3D (W=30)
       ├── Entrenar modelo midiendo tiempo y RAM pico (ResourceProfiler)
       └── Evaluar inferencia sobre X_val_fold (RMSE, MAE, NASA Score, latencia)
6. Evaluación final en Test Set oficial (último ciclo de cada motor)
7. Comparación estadística multimodelo (Friedman + Nemenyi / RM-ANOVA)
```

### 2. Validación Cruzada con k=10 Folds
- **Justificación:** FD001 cuenta con 100 motores de entrenamiento, lo que permite exactamente 10 motores independientes por fold.
- **Potencia Estadística:** 10 bloques pareados incrementan la potencia de las pruebas no paramétricas (Friedman y post-hoc Nemenyi del Issue #8) reduciendo el umbral de diferencia crítica frente a k=5.

### 3. Estructura Jerárquica de Tracking en MLflow
- **Experimento:** `rul_cmapss_FD001` (apuntando a `sqlite:///mlflow.db`).
- **Runs por Modelo:** Registra tags (`model_type`, `subset`), parámetros (`rul_max`, `window_size`, `cv_folds`, hiperparámetros del modelo), métricas promedio y desviación estándar de CV (`cv_rmse_mean`, `cv_mae_mean`, `cv_nasa_mean`, `cv_train_time_mean`, `cv_ram_train_peak_mb`), más el desempeño en test set.
- **Run Multimodelo (Omnibus):** Consolida los vectores de error de todos los modelos y registra como artefacto el archivo `statistical_analysis.json` generado por `compare_multiple_models().to_dict()`.

### 4. Perfilado de Recursos Integrado
Utiliza el context manager `profile_resource_usage()` (`src/evaluation/metrics.py`) para capturar de manera no invasiva:
- Tiempo de entrenamiento (segundos)
- Latencia de inferencia (milisegundos / motor)
- RAM pico en entrenamiento e inferencia (MB)

### 5. CLI Flexible y Parametrizada
El script `scripts/train_eval.py` admitirá ejecución por consola:
```bash
python scripts/train_eval.py --config configs/config_FD001.yaml
python scripts/train_eval.py --config configs/config_FD001.yaml --models rf xgboost mlp
python scripts/train_eval.py --config configs/config_FD001.yaml --dry-run
```

### A. Orquestación y CLI (`scripts/train_eval.py`)
- [ ] Script ejecutable desde consola que carga `configs/config_FD001.yaml`.
- [ ] Argumentos CLI soportados: `--config`, `--models` (filtro opcional de modelos), `--dry-run` (prueba rápida con 2 folds y datos reducidos).
- [ ] Fijación determinista de semilla aleatoria desde configuración (`random_seed: 42`).

### B. Pipeline de Datos y Características
- [ ] Integración de `load_cmapss` para lectura de datos.
- [ ] Filtrado de sensores constantes y cálculo de Piecewise RUL (125 ciclos).
- [ ] Extracción condicional de features según configuración (`rolling_stats`, `trends`).
- [ ] Normalización `MinMaxScaler` ajustada exclusivamente sobre el conjunto de entrenamiento de cada fold.
- [ ] Generación de ventanas temporales 3D `(N, W=30, F)` compatibles con modelos clásicos y de Deep Learning.

### C. Validación y Perfilado de Modelos
- [ ] Validación cruzada `GroupKFold(n_splits=10)` agrupada por `unit_number`.
- [ ] Cálculo de métricas: `rmse`, `mae` y `nasa_score`.
- [ ] Perfilado de tiempo y consumo de memoria RAM pico por fold y por fase.
- [ ] Compatibilidad comprobada con los modelos de la línea base:
  - Clásicos: Random Forest, XGBoost, LightGBM, SVR.
  - Deep Learning: MLP, CNN-1D, LSTM.

### D. Inferencia Estadística y MLflow Tracking
- [ ] Conexión y registro en base de datos local `sqlite:///mlflow.db`.
- [ ] Registro individual de runs con parámetros, métricas de precisión y métricas de eficiencia computacional.
- [ ] Ejecución de `compare_multiple_models()` con los resultados pareados por fold de todos los modelos evaluados.
- [ ] Exportación del artefacto `statistical_analysis.json` con la matriz de comparaciones post-hoc y los rankings al run principal de MLflow.

### E. Verificación
- [ ] Ejecución exitosa de extremo a extremo para al menos 3 modelos base (ej. Random Forest, MLP, LSTM).
- [ ] Verificación de métricas, parámetros y artefactos legibles en `mlflow ui`.

## Out of Scope
- Optimización y tuning bayesiano de hiperparámetros.
- Ejecución sobre subconjuntos multirégimen (FD002, FD003, FD004).
- Modelado con Memoria Asociativa (se evaluará una vez asentados los baselines).
- Generación de tablas en sintaxis LaTeX.

## Implementation Order
1. Creación de `scripts/train_eval.py` con parsing de argumentos CLI y carga de YAML.
2. Implementación de funciones de preparación de datos por fold (anti-leakage) con generación de ventanas 3D.
3. Implementación del bucle de validación cruzada con perfilado de recursos.
4. Integración del cliente MLflow para logging de runs individuales por modelo.
5. Integración del análisis estadístico multimodelo y exportación de `statistical_analysis.json`.
6. Validación y ejecución end-to-end con verificación en la interfaz gráfica de MLflow.
